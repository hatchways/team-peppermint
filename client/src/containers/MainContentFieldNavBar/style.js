import { makeStyles, withStyles } from "@material-ui/core/styles";
import Badge from "@material-ui/core/Badge";

export const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    minHeight: "10vh",
    backgroundColor: "#fff",
    boxShadow: " 0 15px 15px #f5f7fb",
  },
  leftRightSideStyle: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    padding: 20,
  },
  onOfflineStyle: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    width: 50,
  },
  typography: {
    fontSize: "0.9rem",
    fontWeight: 600,
    color: "#3A8DFF",
  },
  [`@media (max-width:510px)`]: {
    root: {
      flexDirection: "column",
      justifyContent: "center",
      padding: 20,
    },
    leftRightSideStyle: {
      padding: 0,
    },
    typography: {
      fontSize: "1.3rem",
    },
  },
}));

export const StyledBadge = withStyles((theme) => ({
  badge: {
    backgroundColor: "#44b700",
    color: "#44b700",
    boxShadow: `0 0 0 2px ${theme.palette.background.paper}`,
    width: theme.spacing(1.1),
    height: theme.spacing(1.1),
    borderRadius: "50%",
    "&::after": {
      position: "absolute",
      top: 0,
      left: 0,
      width: "100%",
      height: "100%",
      borderRadius: "50%",
      animation: "$ripple 1.2s infinite ease-in-out",
      border: "1px solid currentColor",
      content: '""',
    },
  },
  "@keyframes ripple": {
    "0%": {
      transform: "scale(.8)",
      opacity: 1,
    },
    "100%": {
      transform: "scale(2.4)",
      opacity: 0,
    },
  },
}))(Badge);
