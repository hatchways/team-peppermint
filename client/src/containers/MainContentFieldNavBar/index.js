import React, { useState, useContext } from "react";
import { Typography, Switch, ButtonBase } from "@material-ui/core";
import { useStyles, StyledBadge } from "./style";
import { MoreHoriz } from "@material-ui/icons";
import { useUserState } from "../../context/user/userContext";
import ToggleLanguage from "../../context/ToggleLanguage";

const MainContentFieldNavBar = () => {
  const [checked, setChecked] = useState(false);
  const classes = useStyles();
  const toggleLanguageContext = useContext(ToggleLanguage);
  const { user } = useUserState();

  const handleChange = () => {
    setChecked(!checked);
    toggleLanguageContext.setOriginal(!checked);
  };

  return (
    <div className={classes.root}>
      <div className={classes.leftRightSideStyle}>
        <Typography
          variant="h6"
          style={{ fontWeight: 600, marginLeft: 20, marginRight: 20 }}
        >
          {user.name}
        </Typography>
        <div className={classes.onOfflineStyle}>
          <StyledBadge overlap="circle" variant="dot">
            <span></span>
          </StyledBadge>
          <Typography variant="body2" style={{ fontSize: "0.8rem" }}>
            Online
          </Typography>
        </div>
      </div>
      <div className={classes.leftRightSideStyle}>
        {checked ? (
          <Typography
            variant="body2"
            style={{
              fontSize: "0.9rem",
              fontWeight: 600,
              color: "#3A8DFF",
            }}
          >
            Original language
          </Typography>
        ) : (
          <Typography
            variant="body2"
            style={{
              fontSize: "0.9rem",
              fontWeight: 600,
              opacity: 0.5,
            }}
          >
            Original language
          </Typography>
        )}
        <Switch
          checked={checked}
          onChange={handleChange}
          color="primary"
          name="checked"
          inputProps={{ "aria-label": "secondary checkbox" }}
        />
        <ButtonBase
          onClick={() => console.log("Clicked")}
          style={{ marginLeft: 20 }}
        >
          <MoreHoriz />
        </ButtonBase>
      </div>
    </div>
  );
};

export default MainContentFieldNavBar;
