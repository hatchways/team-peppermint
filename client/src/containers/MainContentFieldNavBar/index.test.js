import React from "react";
import MainContentFieldNavBar from "./index";

jest.mock("../../context/user/userContext.js", () => {
  return {
    useUserState: jest.fn(() => ({
      token: "123",
      user: {
        id: "1",
        name: "Jane",
        email: "jane@gmail.com",
        language: "English",
        pictureURL: { name: "123455", url: "https://www.picture.com" },
      },
    })),
  };
});

let wrapper;

beforeEach(() => {
  wrapper = shallow(<MainContentFieldNavBar />);
});

describe("<MainContentFieldNavBar />", () => {
  it("match to snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
