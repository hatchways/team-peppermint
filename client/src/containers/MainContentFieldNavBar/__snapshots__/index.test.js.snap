// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContentFieldNavBar /> match to snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="makeStyles-leftRightSideStyle-2"
  >
    <WithStyles(ForwardRef(Typography))
      style={
        Object {
          "fontWeight": 600,
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
      variant="h6"
    >
      Jane
    </WithStyles(ForwardRef(Typography))>
    <div
      className="makeStyles-onOfflineStyle-3"
    >
      <WithStyles(WithStyles(ForwardRef(Badge)))
        overlap="circle"
        variant="dot"
      >
        <span />
      </WithStyles(WithStyles(ForwardRef(Badge)))>
      <WithStyles(ForwardRef(Typography))
        style={
          Object {
            "fontSize": "0.8rem",
          }
        }
        variant="body2"
      >
        Online
      </WithStyles(ForwardRef(Typography))>
    </div>
  </div>
  <div
    className="makeStyles-leftRightSideStyle-2"
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-typography-4"
      style={
        Object {
          "color": "grey",
          "opacity": 0.5,
        }
      }
      variant="body2"
    >
      Original language
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Switch))
      checked={false}
      color="primary"
      inputProps={
        Object {
          "aria-label": "secondary checkbox",
        }
      }
      name="checked"
      onChange={[Function]}
    />
    <WithStyles(ForwardRef(ButtonBase))
      onClick={[Function]}
      style={
        Object {
          "marginLeft": 20,
        }
      }
    >
      <MoreHorizIcon />
    </WithStyles(ForwardRef(ButtonBase))>
  </div>
</div>
`;
